File: App/src/Adaptive.ReactiveTrader.Client.Domain/Models/CurrencyPairUpdate.cs
using Adaptive.ReactiveTrader.Client.Domain.Repositories;

namespace Adaptive.ReactiveTrader.Client.Domain.Models
{
    public class CurrencyPairUpdate : ICurrencyPairUpdate
    {
        public CurrencyPairUpdate(UpdateType updateType, ICurrencyPair currencyPair)
        {
            UpdateType = updateType;
            CurrencyPair = currencyPair;
        }

        public UpdateType UpdateType { get; private set; }
        public ICurrencyPair CurrencyPair { get; private set; }
    }
}

File: App/src/Adaptive.ReactiveTrader.Client.JS/Models/ReferenceData/CurrencyPairUpdate.ts
class CurrencyPairUpdate implements ICurrencyPairUpdate {
    private _updateType: UpdateType;
    private _currencyPair: ICurrencyPair;

    constructor(updateType: UpdateType, currencyPair: ICurrencyPair) {
        this._currencyPair = currencyPair;
        this._updateType = updateType;
    }

    public get updateType() {
        return this._updateType;
    }

    public get currencyPair() {
        return this._currencyPair;
    }
} 